src/core_plugins/input_control_vis/public/control/filter_manager/phrase_filter_manager.js

```diff
@@ -61,10 +61,22 @@ export class PhraseFilterManager extends FilterManager {
       return;
     }
 
-    return kbnFilters
+    const values = kbnFilters
       .map((kbnFilter) => {
         return this._getValueFromFilter(kbnFilter);
       })
+      .filter(value => {
+        if (typeof value === 'undefined' || value == null) {
+          return false;
+        }
+        return true;
+      });
+
+    if (values.length === 0) {
```

Comment:
This if is unnecessary, as the filter will mean that reduce will be operating over an empty array in the test case, which should be just fie.

Reply:
If the filter removes all items from the array then reduce would return `[]`. The function should return undefined if there are no selected options.